internal/fs/wrappers/monitoring.go

```diff
@@ -18,31 +18,56 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"log"
 	"syscall"
+	"time"
 
+	"github.com/googlecloudplatform/gcsfuse/internal/logger"
 	"github.com/googlecloudplatform/gcsfuse/internal/monitor/tags"
 	"github.com/jacobsa/fuse/fuseops"
 	"github.com/jacobsa/fuse/fuseutil"
+	"go.opencensus.io/plugin/ochttp"
 	"go.opencensus.io/stats"
 	"go.opencensus.io/stats/view"
 	"go.opencensus.io/tag"
 )
 
+var errorLogger *log.Logger
+
 var (
-	opCount = stats.Int64("fs/ops_count", "The number of ops processed by the file system.", stats.UnitDimensionless)
+	opsCount      = stats.Int64("fs/ops_count", "The number of ops processed by the file system.", stats.UnitDimensionless)
+	opsLatency    = stats.Float64("fs/ops_latency", "The latency of a file system operation.", stats.UnitMilliseconds)
+	opsErrorCount = stats.Int64("fs/ops_error_count", "The number of errors generated by file system operation.", stats.UnitDimensionless)
 )
 
 // Initialize the metrics.
 func init() {
+
+	errorLogger = logger.NewError("")
+
+	// Register the view.
 	if err := view.Register(
 		&view.View{
 			Name:        "fs/ops_count",
-			Measure:     opCount,
+			Measure:     opsCount,
 			Description: "The cumulative number of ops processed by the file system.",
 			Aggregation: view.Sum(),
+			TagKeys:     []tag.Key{tags.FSOp},
+		},
+		&view.View{
+			Name:        "fs/ops_error_count",
```

Comment:
Do we get a breakdown of the number of errors by ops type also? i.e. reads got 10 errors, writes got 3, stat got 5 etc ? or this is a total number?

Reply:
yes